### PR TITLE
Add Iron distribution as pre-release

### PR DESCRIPTION
**Pull request by Yadunund**
This PR adds the `distribution.yaml` file for `iron` generated by running the migration tool with the command
```bash
python3 migration-tools/migrate-rosdistro.py --dest iron --source rolling --source-ref d7676d5c2ce5a99abb43ac2a3e6b4510fe50a261 --release-org ros2-gbp
```
with `source-ref` being the latest commit to `master` at the time of run.

Some repositories were manually migrated due to issues encountered during the first run.

### CODE REVIEW

File: iron/distribution.yaml [hunk 1]
```diff
@@ -0,0 +1,6952 @@
...
+      test_pull_requests: true
+      type: git
+      url: https://github.com/ros/ros_tutorials.git
+      version: rolling
+    status: maintained
+  tuw_geometry:
+    doc:
+      type: git
+      url: https://github.com/tuw-robotics/tuw_geometry.git
+      version: humble
```
Comment (Yadunund) [inline iron/distribution.yaml:6258]: CI is failing as `humble` branch does not exist at `https://github.com/tuw-robotics/tuw_geometry.git`. These `humble` instances should be changed to `ros2`.
```suggestion
      version: ros2
```

Comment (Yadunund) [reply iron/distribution.yaml]: [3f74403](https://github.com/ros/rosdistro/pull/36923/commits/3f74403baddf0b50d176730a167c97b448e3f24b)

**Review by cottsay — APPROVED** [review]
:metal: :turtle: :robot:

**Review by clalancette — APPROVED** [review]
Woohoo!

**Pull request merged**